src/clients/HubPoolClient.ts

```diff
@@ -347,7 +347,7 @@ export class HubPoolClient extends BaseAbstractClient {
     chain: number,
     chainIdListOverride?: number[]
   ): number | undefined {
-    const chainIdList = chainIdListOverride ?? this.configStoreClient.getEnabledChains(latestMainnetBlock);
+    const chainIdList = chainIdListOverride ?? this.configStoreClient.getChainIdIndicesForBlock(latestMainnetBlock);
```

Comment:
Should we remove the `chainIdListOverride` entirely in favor of enforcing the on-chain call?

Reply:
Yea in prod we'd never override it but it might be useful for tests. Wdyt how do the tests look without this override option?

Reply:
Yeah we should keep it in the end. A lot of things break and we'd need addtl functionality to get to a place where we are currently.